chain/client/src/rocksdb_metrics.rs

```diff
@@ -0,0 +1,162 @@
+use near_metrics::{try_create_gauge_vec, try_create_int_gauge};
+use prometheus::{GaugeVec, IntGauge};
+use std::collections::HashMap;
+
+#[derive(Default)]
+/// Creates prometheus metrics on-demand for exporting RocksDB statistics.
+pub(crate) struct RocksDBMetrics {
+    // Contains counters and sums, which are integer statistics in RocksDB.
+    int_gauges: HashMap<String, IntGauge>,
+    // Contains floating point statistics, such as quantiles of timings.
+    gauges: HashMap<String, GaugeVec>,
+}
+
+impl RocksDBMetrics {
+    pub fn export_stats_as_metrics(
+        &mut self,
+        stats: &[(&str, Vec<StatsValue>)],
+    ) -> anyhow::Result<()> {
+        for (stat_name, values) in stats {
+            if values.len() == 1 {
```

Comment:
Can fold the two conditions I think:

`if let [StatsValue::Count(value)] = values.as_slice()`